tests/gui_core/test_context_tabular_data_edit.py

```diff
@@ -0,0 +1,388 @@
+
+    def __call_update_data(self, col=None, idx=None, new_value=None):
+        mockGui = Mock(Gui)
+        mockGui._get_client_id = lambda: "a_client_id"
+        gui_core_context = _GuiCoreContext(mockGui)
+        payload = {"user_data": {"dn_id": dn.id}, "error_id": "error_var"}
+        if idx is not None:
+            payload["index"] = idx
+        if col is not None:
+            payload["col"] = col
+        if new_value is not None:
+            payload["value"] = new_value
+        assign = Mock()
+        gui_core_context.tabular_data_edit(
+            state=MockState(assign=assign),
+            var_name="",
+            payload=payload,
+        )
+        return assign
+
+    def test_edit_pandas_wrong_idx(self):
+        data = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+        dn.write(data)
+        idx = 5
+        col = "a"
+        new_value = 100
+        new_data = data.copy()
+        new_data.at[idx, col] = new_value
+        with patch("taipy.gui_core._context.is_readable", side_effect=is_true):
+            with patch("taipy.gui_core._context.is_editable", side_effect=is_true):
+                with patch("taipy.gui_core._context.core_get", side_effect=core_get) as mock_core_get:
+                    with patch.object(DataNode, "write") as mock_write:
+
+                        assign = self.__call_update_data(col, idx, new_value)
+
+                        mock_core_get.assert_called_once_with(dn.id)
+                        mock_write.assert_called_once()
+                        # Cannot use the following line because of the pandas DataFrame comparison
+                        # mock_write.assert_called_once_with(new_data, editor_id="a_client_id", comment=None
+                        # Instead, we will compare the arguments of the call manually
+                        assert mock_write.call_args_list[0].args[0].equals(new_data)
+                        assert mock_write.call_args_list[0].kwargs["editor_id"] == "a_client_id"
+                        assert mock_write.call_args_list[0].kwargs["comment"] is None
+                        assign.assert_called_once_with("error_var", "")
+
+    def test_edit_pandas_wrong_col(self):
+        data = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+        dn.write(data)
+        idx = 0
+        col = "c"
+        new_value = 100
+        new_data = data.copy()
+        new_data.at[idx, col] = new_value
+        with patch("taipy.gui_core._context.is_readable", side_effect=is_true):
+            with patch("taipy.gui_core._context.is_editable", side_effect=is_true):
+                with patch("taipy.gui_core._context.core_get", side_effect=core_get) as mock_core_get:
+                    with patch.object(DataNode, "write") as mock_write:
+                        assign = self.__call_update_data(col, idx, new_value)
+                        mock_core_get.assert_called_once_with(dn.id)
+                        mock_write.assert_called_once()
```

Comment:
I would expect the function to fail as the column is incorrect. What do you think @FredLL-Avaiga ?

Reply:
how could the column be incorrect ?

Reply:
that can't happen with the UI components